tests/epyccel/recognised_functions/test_numpy_funcs.py

```diff
@@ -5953,3 +5953,301 @@ def get_prod(arr):
+    integer32 = randint(min_int32, max_int32, size=size, dtype=np.int32)
+    integer64 = randint(min_int64, max_int64, size=size, dtype=np.int64)
+
+    fl = uniform(low=-((abs(min_float) / (size[0] * size[1]))**(1/2)), high=(abs(max_float) / (size[0] * size[1]))**(1/2), size=size)
+    fl32 = uniform(low=-((abs(min_float32) / (size[0] * size[1]))**(1/2)), high=(abs(max_float32) / (size[0] * size[1]))**(1/2), size=size)
+    fl32 = np.float32(fl32)
+    fl64 = uniform(low=-((abs(min_float64) / (size[0] * size[1]))**(1/2)), high=(abs(max_float64) / (size[0] * size[1]))**(1/2), size=size)
+
+    cmplx128_from_float32 = uniform(low=-((abs(min_float32) / (size[0] * size[1] * 2))**(1/2)), high=(abs(max_float32) / (size[0] * size[1] * 2))**(1/2), size=size) + uniform(low=-((abs(min_float32) / (size[0] * size[1] * 2))**(1/2)), high=(abs(max_float32) / (size[0] * size[1] * 2))**(1/2), size=size) * 1j
+    # the result of the last operation is a Python complex type which has 8 bytes in the alignment,
+    # that's why we need to convert it to a numpy.complex64 the needed type.
+    cmplx64 = np.complex64(cmplx128_from_float32)
+    cmplx128= uniform(low=-((abs(min_float64) / (size[0] * size[1] * 2))**(1/2)), high=(abs(max_float64) / (size[0] * size[1] * 2))**(1/2), size=size) + uniform(low=-((abs(min_float64) / (size[0] * size[1] * 2))**(1/2)), high=(abs(max_float64) / (size[0] * size[1] * 2))**(1/2), size=size) * 1j
+
+    f_bl = epyccel(get_norm, language=language)
+
+    assert np.isclose(f_bl(bl), get_norm(bl), rtol=rtol, atol=atol)
+
+    f_integer8 = epyccel(get_norm, language=language)
+    f_integer16 = epyccel(get_norm, language=language)
+    f_integer = epyccel(get_norm, language=language)
+    f_integer32 = epyccel(get_norm, language=language)
+
+    assert np.isclose(f_integer8(integer8), get_norm(integer8), rtol=rtol, atol=atol)
+    assert np.isclose(f_integer16(integer16), get_norm(integer16), rtol=rtol, atol=atol)
+    assert np.isclose(f_integer(integer), get_norm(integer), rtol=rtol, atol=atol)
+    assert np.isclose(f_integer32(integer32), get_norm(integer32), rtol=rtol, atol=atol)
+
+    f_integer64 = epyccel(get_norm, language=language)
+    assert np.isclose(f_integer64(integer64), get_norm(integer64), rtol=rtol, atol=atol)
+
+    f_fl = epyccel(get_norm, language=language)
+    f_fl32 = epyccel(get_norm, language=language)
+    f_fl64 = epyccel(get_norm, language=language)
+
+    assert np.isclose(f_fl(fl), get_norm(fl), rtol=rtol, atol=atol)
+    assert np.isclose(f_fl32(fl32), get_norm(fl32), rtol=rtol, atol=atol)
+    assert np.isclose(f_fl64(fl64), get_norm(fl64), rtol=rtol, atol=atol)
+
+    f_complex64 = epyccel(get_norm, language=language)
+    f_complex128 = epyccel(get_norm, language=language)
+
+    assert np.isclose(f_complex64(cmplx64), get_norm(cmplx64), rtol=1e-6, atol=1e-7)
+    assert np.isclose(f_complex128(cmplx128), get_norm(cmplx128), rtol=rtol, atol=atol)
```

Comment:
Why do you specify `rtol` and `atol` on line 6252 but use the variables on line 6253?

Reply:
sorry, i missed that.